homeassistant/components/zabbix/__init__.py

```diff
@@ -68,101 +97,292 @@
+            floats[entity_id] = _state_as_value
+        except ValueError:
+            strings[entity_id] = state.state
+    for key, value in state.attributes.items():
+        # For each value we try to cast it as float
+        # But if we cannot do it we store the value
+        # as string
+        attribute_id = f"{entity_id}/{key}"
+        try:
+            float_value = float(value)
+        except (ValueError, TypeError):
+            float_value = None
+        if float_value is None or not math.isfinite(float_value):
+            strings[attribute_id] = str(value)
+        else:
+            floats[attribute_id] = float_value
+    float_keys_count = len(float_keys)
+    float_keys.update(floats.keys())
+    if len(float_keys) != float_keys_count:
+        floats_discovery = [
+            {"{#KEY}": str(float_key)[:230]} for float_key in float_keys
+        ]
+        metric = ItemValue(
+            publish_states_host,
+            "homeassistant.floats_discovery",
+            json.dumps(floats_discovery),
+        )
+        metrics.append(metric)
+    for key, value in floats.items():
+        metric = ItemValue(
+            publish_states_host, f"homeassistant.float[{str(key)[:230]}]", value
+        )
+        metrics.append(metric)
 
-    conf = config[DOMAIN]
-    protocol = "https" if conf[CONF_SSL] else "http"
+    string_keys.update(strings)
 
-    url = urljoin(f"{protocol}://{conf[CONF_HOST]}", conf[CONF_PATH])
-    username = conf.get(CONF_USERNAME)
-    password = conf.get(CONF_PASSWORD)
+    return metrics
 
-    publish_states_host = conf.get(CONF_PUBLISH_STATES_HOST)
 
-    entities_filter = convert_include_exclude_filter(conf)
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Zabbix component from a ConfigEntry."""
+    zabbix_sender: AsyncSender
+    host: str
+    filter_dict: dict[str, list[str]] = {}
+    hass_data: dict
+    instance: ZabbixThread
 
-    try:
-        zapi = ZabbixAPI(url=url, user=username, password=password)
-        _LOGGER.info("Connected to Zabbix API Version %s", zapi.api_version())
-    except ZabbixAPIException as login_exception:
-        _LOGGER.error("Unable to login to the Zabbix API: %s", login_exception)
-        return False
-    except HTTPError as http_error:
-        _LOGGER.error("HTTPError when connecting to Zabbix API: %s", http_error)
-        zapi = None
-        _LOGGER.error(RETRY_MESSAGE, http_error)
-        event_helper.call_later(
-            hass,
-            RETRY_INTERVAL,
-            lambda _: setup(hass, config),  # type: ignore[arg-type,return-value]
-        )
+    hass.data.setdefault(DOMAIN, {})
+    # Already configured from configuration.yaml. Not allow UI config.
+    if hass.data[DOMAIN].get(NEW_CONFIG) is False:
         return True
 
-    hass.data[DOMAIN] = zapi
+    # Continue with the new config format
+    hass_data = dict(entry.data)
```

Comment:
all the places where you have access to `hass.data` you also have access to `entry` so I don't see a point in storing it there

Reply:
Removed hass_data and directly use entry.dict instead.
Still I store current entry under hass.data for easy access from other sensors.py